src/core/commit.rs

```diff
@@ -220,28 +211,6 @@ impl<J: Jet> CommitNode<J> {
         self.iter().count()
     }
 
-    /// Create a node from its underlying combinator.
-    fn node_from_inner(
-        context: &mut Context<J>,
-        inner: CommitNodeInner<J>,
-        _left: Option<Rc<CommitNode<J>>>,
-        _right: Option<Rc<CommitNode<J>>>,
-        hint: &'static str,
-    ) -> Result<Rc<CommitNode<J>>, Error> {
-        match UnificationArrow::for_node(context, &inner) {
-            Ok(arrow) => Ok(Rc::new(CommitNode {
-                cmr: Cmr::compute(&inner),
-                inner,
-                arrow,
-            })),
-            Err(Error::Unification(unification_hint)) => Err(Error::TypeCheck {
-                unification_hint,
-                root_hint: hint,
-            }),
-            Err(error) => Err(error),
-        }
-    }
```

Comment:
We remove this because the hints from the type inference already explain the error, plus we want to third parties like your decoder to display helpful error messages to the user?

Reply:
Correct. Adding a second fixed hint provides pretty little value in my opinion.

What we may do in the future is change the `hint` field to an owned `String` or even a closure that can take extra auxiliary information.... but for now my decoder is able to get a fair bit of mileage out of the single fixed hint and a line/column.

Anyway improving the errors is for a later PR.